src/utility/utility.cpp

```diff
@@ -81,10 +81,52 @@ Transpose(const std::vector<std::vector<double>> &A)
   return r;
 }
 
+double Li2(const double &x)
+{
+  if (x == 0) return 0;
+  if (x == 1) return pow(M_PI, 2) / 6.;
+  if (x < -1) return -pow(M_PI, 2) / 6. - pow(log(-x), 2) / 2. - Li2(1. / x);
+  if (x < 0) return 1 / 2. * Li2(-x * -x) - Li2(-x);
+  if (x > 0.5) return pow(M_PI, 2) / 6. - log(x) * log(1 - x) - Li2(1 - x);
+  double sum = 1e-100;
```

Comment:
```suggestion
  double sum = 0;
```

Why not start with 0?

Reply:
Prevent a 1/0 on line 94.

Reply:
Why not test for it and have a proper case for it?

Reply:
I didn't want an additional if statement inside the loop.
I found a middle ground. The sum is initialized at ```sum = x```, which is the first element of the sum at ```k=1```, then the sum loop starts at ```k=2```. I commented on this logic for posterity.

Reply:
What prevents it from becoming zero between the loop?

Reply:
By the [properties](https://en.wikipedia.org/wiki/Dilogarithm#Identities) of the dilogarithm, if $x<0$ then recall it recursively with positive arguments. Therefore, in the sum $x>0$ always so
$$\sum_{k=2}^\infty\frac{x^k}{k^2}$$
is a series of positive numbers and it can never vanish.